From 55c2e322a21f39cc4b70e87fbac835cceb9ac161 Mon Sep 17 00:00:00 2001
From: Manpreet <manpreet.singh@mediatek.com>
Date: Mon, 17 Dec 2018 16:28:59 +0530
Subject: [PATCH 37/45] Null locale causing the VTS fail.

Added a null check to avoid exceptions caused by timing delay during VTS run.This case fail is because AccessibilityWindowQueryActivity execute scheduleTraversals before Activity destroyed and execute doTraversal after Activity destroyed. This means window animation is not stop after AccessibilityWindowQueryActivity is destroyed and this will lead to test crash.

Test: Tested 5 times and it is passed 5 times.
Bug: 121169721
Change-Id: I915d46f925fb7a5ca43cac3ef3a343822b952b02
(cherry picked from commit b99c60c36e095a597015b5e3b738026148f27349)
---
 core/java/android/view/AccessibilityIterators.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/java/android/view/AccessibilityIterators.java b/core/java/android/view/AccessibilityIterators.java
index ca54bef1e78..c8362600876 100644
--- a/core/java/android/view/AccessibilityIterators.java
+++ b/core/java/android/view/AccessibilityIterators.java
@@ -145,6 +145,9 @@ public final class AccessibilityIterators {
         @Override
         public void onConfigurationChanged(Configuration globalConfig) {
             final Locale locale = globalConfig.getLocales().get(0);
+            if (locale == null) {
+                return;
+            }
             if (!mLocale.equals(locale)) {
                 mLocale = locale;
                 onLocaleChanged(locale);
-- 
2.17.1

